openraft/src/core/client.rs

```diff
@@ -60,7 +47,7 @@ impl<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> LeaderS
     pub(super) async fn handle_check_is_leader_request(&mut self, tx: RaftRespTx<(), CheckIsLeaderError<C::NodeId>>) {
         // Setup sentinel values to track when we've received majority confirmation of leadership.
 
-        let mem = &self.core.effective_membership.membership;
+        let mem = &self.core.engine.state.effective_membership.membership;
```

Comment:
this pr has too many variable with `self.core.engine.state.effective_membership.membership`, make the code line too long to read.

may be add some methods in engine like `get_effective_membership` to return this filed?

Reply:
and IMHO, one good interface has not to make the user care about the inner field too much, user and interface communicate with methods, not field.

Reply:
when communicate with methods, the next time the field member change(such as rename,etc),there wiil be no more rename change like this pr.

and other benefit with communicating with methods is force the author thinking about the API or function this interface provided.

Reply:
`core.engine.state` will be removed soon.
Eventually `RaftCore` will become a pure **runtime** and it accesses the **engine** through methods provided by `Engine`.

But for now, just directly access these fields. Otherwise, this PR will get too many lines of changes involved.